lib/qbwc/request/products.rb

```diff
@@ -138,6 +138,30 @@ def polling_current_items_xml(params, config)
               #{query_by_date(params, time)}
               <!-- <IncludeRetElement>Name</IncludeRetElement> -->
             </ItemInventoryAssemblyQueryRq>
+            <!-- polling non inventory products -->
+            <ItemNonInventoryQueryRq requestID="#{session_id}">
+            <MaxReturned>1000</MaxReturned>
+              #{query_by_date(params, time)}
+              <!-- <IncludeRetElement>Name</IncludeRetElement> -->
+            </ItemNonInventoryQueryRq>
+            <!-- polling sales tax products -->
+            <ItemSalesTaxQueryRq requestID="#{session_id}">
+            <MaxReturned>1000</MaxReturned>
+              #{query_by_date(params, time)}
+              <!-- <IncludeRetElement>Name</IncludeRetElement> -->
+            </ItemSalesTaxQueryRq>
+            <!-- polling service products -->
+            <ItemServiceQueryRq requestID="#{session_id}">
+            <MaxReturned>1000</MaxReturned>
+              #{query_by_date(params, time)}
+              <!-- <IncludeRetElement>Name</IncludeRetElement> -->
+            </ItemServiceQueryRq>
+            <!-- polling discount products -->
+            <ItemDiscountQueryRq requestID="#{session_id}">
+            <MaxReturned>1000</MaxReturned>
+              #{query_by_date(params, time)}
+              <!-- <IncludeRetElement>Name</IncludeRetElement> -->
+            </ItemDiscountQueryRq>
```

Comment:
if we allow max of 1000 for each we could return more than 4000 at once, is that fine ?

Reply:
Hmm - good point. I didn't even think about this. Probably not. Any idea why it was originally set to 1000 each? I'd say we drop that down to 25 or something. Wdyt?